clio-server/src/main/scala/org/broadinstitute/clio/server/service/SchemaService.scala

```diff
@@ -0,0 +1,188 @@
+package org.broadinstitute.clio.server.service
+
+import java.time.OffsetDateTime
+
+import io.circe._
+import org.broadinstitute.clio.transfer.model.TransferReadGroupV1QueryOutput
+import shapeless._
+import shapeless.labelled._
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.reflect.runtime.universe.{Type, TypeTag, typeOf, typeTag}
+
+/**
+  * Generate JSON schema from case classes such as the QueryOutputs.
+  *
+  * JsonSchema uses the same shapeless pattern as the
+  * org.broadinstitute.clio.util.generic.FieldMapper
+  */
+object SchemaService {
+
+  /**
+    * a JSON schema for A suitable for Circe
+    *
+    * @tparam A is a case class such as a QueryOutput
+    */
+  trait JsonSchema[A] {
+
+    /**
+      * @return the names of fields required in A
+      */
+    def required: Set[String]
+
+    /**
+      * @return the JSON properties of all fields in A
+      */
+    def properties: Map[String, Json]
+
+    /**
+      * @return a Circe representation of the schema
+      */
+    def toJson: Json
+  }
+
+  object JsonSchema {
+
+    /**
+      * Summon a JsonSchema encoder from implicit scope.
+      *
+      * @param jsonSchema to summon
+      * @tparam A is a case class such as a QueryOutput
+      * @return the implicit JsonSchema object
+      */
+    def apply[A](implicit jsonSchema: JsonSchema[A]): JsonSchema[A] =
+      jsonSchema
+
+    /**
+      * Specialize the JsonSchema trait to combine theRequired and
+      * theProperties into a representation of a JsonSchema for A.
+      *
+      * @param theRequired fields are typed without Option
+      * @param theProperties of the fields describe their types
+      * @tparam A is a case class such as a QueryOutput
+      * @return a new JSON schema representation for A
+      */
+    def makeSchema[A](theRequired: Set[String],
+                      theProperties: Map[String, Json]): JsonSchema[A] =
+      new JsonSchema[A] {
+        override val required: Set[String] = theRequired
+        override val properties: Map[String, Json] = theProperties
+        private val typeObject = "type" -> Json.fromString("object")
+        private val allRequired = "required" -> Json.fromValues(
+          required map Json.fromString
+        )
+        private val allProperties = "properties" -> Json.fromFields(properties)
+        override val toJson: Json =
+          Json.obj(typeObject, allRequired, allProperties)
+      }
+
+    /**
+      * The JSON schema for an empty class is empty.
+      */
+    implicit val hnilSchema: JsonSchema[HNil] =
+      makeSchema(Set.empty, Map.empty)
+
+    /**
+      * type and format pairs for building JSON property schema
```

Comment:
Does the scaladoc apply to both methods? I don't know much about scaladoc but in javadoc you need to have a separate comment for each method.